projects/plugins/jetpack/class.json-api-endpoints.php

```diff
@@ -1297,18 +1297,18 @@ public function user_can_view_post( $post_id ) {
 
 		if ( 'inherit' === $post->post_status ) {
 			$parent_post     = get_post( $post->post_parent );
-			$post_status_obj = get_post_status_object( $parent_post->post_status );
+			$post_status_obj = get_post_status_object( $parent_post->post_status ) ?? null;
 		} else {
 			$post_status_obj = get_post_status_object( $post->post_status );
 		}
 
-		if ( ! $post_status_obj->public ) {
+		if ( ! isset( $post_status_obj->public ) || ! $post_status_obj->public ) {
```

Comment:
Could be more concise here.
```suggestion
		if ( empty( $post_status_obj->public )  ) {
```